File: calc.rb
def add(x, y)
    return x + y
end

def substract(x, y)
    return x - y
end

def multiply(x, y)
    return x * y
end

def divide(x, y)
    return x / y
end

puts "Select operation."
puts "1.Add"
puts "1.Substract"
puts "1.Multiply"
puts "1.Divide"

loop do
    print "Enter choice(1/2/3/4): "
    choice = gets.chomp

    if %w(1 2 3 4).include?(choice)
        begin
            print "Enter first number: "
            num1 = Float(gets.chomp)
            print "Enter second number: "
            num2 = Float(gets.chomp)
        rescue ArgumentError
            puts "Invalid input, Please enter a number."
            next
        end

        case choice
        when '1'
            puts "#{num1} + #{num2} = #{add(num1, num2)}"
        when '2'
            puts "#{num1} - #{num2} = #{substract(num1, num2)}"
        when '3'
            puts "#{num1} * #{num2} = #{multiply(num1, num2)}"
        when '4'
            puts "#{num1} / #{num2} = #{divide(num1, num2)}"
        end

        print "Let's do next calculation? (yes/no): "
        break if gets.chomp == "no"
    else
        puts "Invalid Input"
    end
end

File: calc.js
function add(x, y) {
	return x + y;
}

function subtract(x, y) {
	return x - y;
}

function multiply(x, y) {
	return x * y;
}

function divide(x, y) {
	return y !== 0 ? x / y : "Division by zero is not allowed";
}

console.log("Select operation.");
console.log("1. Add");
console.log("2. Subtract");
console.log("3. Multiply");
console.log("4. Divide");

while (true) {
	const choice = readline("Enter choice(1/2/3/4): ");

	if (choice === "1" || choice === "2" || choice === "3" || choice === "4") {
		try {
			const num1 = parseFloat(prompt("Enter first number: "));
			const num2 = parseFloat(prompt("Enter second number: "));

			if (choice === "1") {
				console.log(num1, "+", num2, "=", add(num1, num2));
			} else if (choice === "2") {
				console.log(num1, "-", num2, "=", subtract(num1, num2));
			} else if (choice === "3") {
				console.log(num1, "*", num2, "=", multiply(num1, num2));
			} else if (choice === "4") {
				console.log(num1, "/", num2, "=", divide(num1, num2));
			}

			if (prompt("Let's do next calculation? (yes/no): ").toLowerCase() !== "yes") {
				break;
			}
		} catch (error) {
			console.error("Invalid input, Please enter a number.");
		}
	} else {
		console.log("Invalid Input");
	}
}
